src/metabase/models/setting.clj

```diff
@@ -1079,6 +1086,15 @@
        (not= (:setter setting-definition) :none)
        (not (ns-in-test? (:namespace setting-definition)))))
 
+(defn- inherit-base
+  "Recursively uses the values in `:base` as default values for `setting-options`"
+  [{:keys [base] :as setting-options}]
+  (let [without-base (dissoc setting-options :base)]
+    (if-not base
+      without-base
+      (do (assert (map? base))
+          (merge (inherit-base base) without-base)))))
```

Comment:
I don't think this should recurse. It's simpler without it and we have no need for it.
```suggestion
  (let [without-base (dissoc setting-options :base)]
    (merge base without-base))
```
Also I feel the assertion is unnecessary. `merge` with other types should fail

Reply:
> I don't think this should recurse. 

Transitivity is something I would expect from an inheritance mechanism. Since we've gone the route of keeping type as separate concern, and users can still merge their own maps before passing to the macro, I'll go ahead and remove this. Using merge does mean losing safeguards like making read-only "final" etc, but perhaps this is YAGNI.